src/client/components/series-menu/quantile-series-menu.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import { Measure } from "../../../common/models/measure/measure";
+import { SeriesList } from "../../../common/models/series-list/series-list";
+import { QuantileSeries } from "../../../common/models/series/quantile-series";
+import { Series } from "../../../common/models/series/series";
+import { SeriesFormat } from "../../../common/models/series/series-format";
+import { Binary } from "../../../common/utils/functional/functional";
+import { Preset } from "../input-with-presets/input-with-presets";
+import { FormatPicker } from "./format-picker";
+import { QuantilePicker } from "./quantile-picker";
+
+interface QuantileSeriesMenuProps {
+  measure: Measure;
+  initialSeries: Series;
+  series: QuantileSeries;
+  seriesList: SeriesList;
+  onChange: Binary<QuantileSeries, boolean, void>;
+}
+
+const percentiles: Array<Preset<number>> = [
+  { identity: 50, name: "50" },
+  { identity: 75, name: "75" },
+  { identity: 90, name: "90" },
+  { identity: 95, name: "95" },
+  { identity: 99, name: "99" }
+];
+
+export const QuantileSeriesMenu: React.SFC<QuantileSeriesMenuProps> = ({ seriesList, initialSeries, measure, series, onChange }) => {
+
+  function isSeriesValid(series: QuantileSeries) {
+    return validateSeries(series) === null;
+  }
+
+  function validateSeries(series: QuantileSeries): string | null {
```

Comment:
Do not rely on hoisting, please. Move this function above `isSeriesValid`. Please.